Controller/src/main/java/com/umn/adscontroller/service/DataService.java

```diff
@@ -0,0 +1,52 @@
+package com.umn.adscontroller.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class DataService {
+
+    private RestTemplate restTemplate;
+
+    DataService(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    @Value("${controller.server.list}")
+    private String[] arrayOfStrings;
+
+
+    public String getData(String inputKey) {
+        int serverId = computeHash(inputKey);
+
+        String baseUrl = arrayOfStrings[serverId];
+        String url = String.format("http://%s/api/%s", baseUrl, inputKey);
+
+        return restTemplate.getForObject(url, String.class);
+    }
+
+    public void putData(String inputKey, String inputValue) {
+        int serverId = computeHash(inputKey);
+
+        String baseUrl = arrayOfStrings[serverId];
+        String url = String.format("http://%s/api/%s", baseUrl, inputKey);
+
+        restTemplate.put(url, inputValue);
+    }
+
+    private int computeHash(String inputKey) {
```

Comment:
Can we take the number of servers as input, that way we don't need to modify the code every time?

Reply:
Sounds good. I will determine the number based on the number of cache servers given in the properties file